src/ex1/script.js

```diff
@@ -1,50 +1,111 @@
 const input = document.querySelector(".todo-input");
-const addButton = document.querySelector(".todo-button");
+const addButton = document.querySelector(".add-todo-button");
+const deleteAll = document.querySelector(".todo-button-delete-all");
 const todoList = document.querySelector(".todo-list");
+let COUNTER = 0;
 
-addButton.addEventListener("click", () => addTodo(input));
-input.addEventListener("keypress", (e) => {
-  if (e.key === "Enter") {
-    addTodo(input);
+const pokemonClient = new PokemonClient();
+const itemManager = new ItemManager();
+
+class Main {
+  init() {
+    addButton.addEventListener("click", () => addTodo(input));
+    deleteAll.addEventListener("click", () => deleteAllTodos());
+    input.addEventListener("keypress", (e) => {
+      if (e.key === "Enter") {
+        addTodo(input);
+      }
+    });
   }
-});
+}
 
 function addTodo(input) {
   if (input.value.length === 0) {
-    changeInputPlaceholder(false);
+    isInputValid(false);
     return;
   }
-  changeInputPlaceholder(true);
-  const todoItem = document.createElement("div");
-  const newTodo = document.createElement("li");
-  const deleteButton = document.createElement("button");
-  const trashIcon = document.createElement("i");
-
-  newTodo.taskName = input.value;
-  newTodo.innerHTML = newTodo.taskName.substring(0, 20);
-  if (newTodo.taskName.length > 20) {
-    newTodo.innerHTML += "...";
+  isInputValid(true);
+
+  let inputArray = input.value.split(",");
+
+  const isInputContainNaN = [...inputArray].some((item) => {
+    let parseToInt = parseInt(item);
+    return isNaN(parseToInt);
+  });
+  if (isInputContainNaN) {
+    itemManager.addTodo({ name: input.value, id: ++COUNTER, pokemonId: -1 });
+    return render();
   }
+  let promisesArray = inputArray.map((item) => {
+    return pokemonClient.fetchPokemon(item);
+  });
+  Promise.all(promisesArray).then((promiseFullfield) => {
```

Comment:
better to export here to function. its a bit hard to follow when there is so much logic inside the block